solr/core/src/java/org/apache/solr/store/blob/process/CorePullTask.java

```diff
@@ -286,7 +286,7 @@ void pullCoreFromBlob(boolean isLeaderPulling) throws InterruptedException {
       }
 
       // Get local metadata + resolve with blob metadata. Given we're doing a pull, don't need to reserve commit point
-      ServerSideMetadata serverMetadata = new ServerSideMetadata(pullCoreInfo.getCoreName(), storeManager.getCoreContainer(), false);
+      ServerSideMetadata serverMetadata = new ServerSideMetadata(pullCoreInfo.getCoreName(), storeManager.getCoreContainer(), false, true);
```

Comment:
Maybe add a short comment explaining the passed value of captureDirHash. Same comment for BlobStoreUtils.java and CorePusher.java.